modules/core/src/main/java/org/apache/ignite/configuration/IgniteConfiguration.java

```diff
@@ -2629,10 +2661,53 @@ public boolean isAutoActivationEnabled() {
      * @param autoActivation Auto activation enabled flag value.
      * @return {@code this} instance.
      * @see #isAutoActivationEnabled()
+     * @deprecated Use {@link IgniteConfiguration#setClusterStateOnStart(ClusterState)} instead.
      */
+    @Deprecated
     public IgniteConfiguration setAutoActivationEnabled(boolean autoActivation) {
+        U.warn(log, "Property autoActivation deprecated. Use clusterStateOnStart instead.");
+
         this.autoActivation = autoActivation;
 
+        autoActivationPropSetFlag = true;
+
+        return this;
+    }
+
+    /**
+     * Gets state of cluster on start.
+     * <br/>
+     * For <b>in-memory cluster</b> this state will be applied to the first started node. If
+     * cluster state on start is {@link ClusterState#INACTIVE}, there will be no cache partition map exchanges
+     * performed until the cluster is activated. This should significantly speed up large topology startup time.
+     * <br/>
+     * For <b>persistent cluster</b> If state is differ from {@link ClusterState#INACTIVE} and BaselineTopology is set
```

Comment:
What do you mean under "BaselineTopology is set"?
It would be better to explain usability (manual activation is not required) when assembling cluster with non-clear persistence directories.

Reply:
done